### PR TITLE
Add several more flags, bit numbers, and offset constants

### PR DESCRIPTION
**Pull request by rondnelson99**
Solves #13.
Am I doing this right? This is my first time contributing to someone else's project.

### CODE REVIEW

**Comment by ISSOtm** [comment]
Please also update the revision number, and add a line to the history explaining the changes.

**Comment by rondnelson99** [comment]
Cool. I can add those, but there were a couple of other things that I think would also be worthwhile to add. One is adding flag definitions for `rSC` and `rIF`, since they are both missing these. I would also add bit numbers to them as there could be utility in using the `bit` instruction with them. Another thing I feel would be useful would be defining `WX_OFFSET` to 7 to account for the fact that 7 must be added to a screen position to calculate a `rWX` position. 

Are there additions worthwhile? If so, should I just commit them to my fork or should I make separate pull requests for them?

**Comment by rondnelson99** [comment]
Well, I went ahead and commited those changes. I can always revert them if they're undesirable. I'm not too sure about the `rIF` flags myself since they're just copy-pasted from the `rIE` flags.

File: hardware.inc [hunk 1]
```diff
@@ -103,6 +104,13 @@ DEF rSB EQU $FF01
 ; --
 DEF rSC EQU $FF02
 
+DEF SCF_START  EQU %10000000 ;Transfer Start Flag (1=Transfer in progress, or requested)
+DEF SCF_SPEED  EQU %00000010 ;Clock Speed (0=Normal, 1=Fast) ** CGB Mode Only **
+DEF SCF_SOURCE EQU %00000001 ;Shift Clock (0=External Clock, 1=Internal Clock)
+
+DEF SCB_START  EQU $7
```
Comment (ISSOtm) [inline hardware.inc:111]: There's no need for those to be hexadecimal, tbh.

File: hardware.inc [hunk 2]
```diff
@@ -145,6 +153,18 @@ DEF TACF_262KHZ EQU %00000001
 ; --
 DEF rIF EQU $FF0F
 
+DEF IFF_HILO   EQU %00010000 ; Transition from High to Low of Pin number P10-P13
```
Comment (ISSOtm) [inline hardware.inc:156]: There are already covered by `IEF_*`

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: hardware.inc (hunk 1 above)
Comment (rondnelson99) [reply hardware.inc:111]: This is true. I made them hexadecimal to be consistent with the PADB_ set that inspired this pull request, but I'm not sure why they were ever hexadecimal in the first place.

File: hardware.inc (hunk 2 above)
Comment (rondnelson99) [reply hardware.inc:156]: Yeah fair enough. I'll remove them, and move/rename the bit numbers to the IE section.

**Comment by rondnelson99** [comment]
I've additionally added offset constants for OAM X and Y position in the same vein as the WX_OFFSET I committed earlier.

**Review by avivace — APPROVED** [review]
Looks ok to me! Thanks @rondnelson99 🥳

File: hardware.inc [hunk 3]
```diff
@@ -541,6 +555,7 @@ DEF rWY EQU $FF4A
 ; -- Values of 0-6 and 166 are unreliable due to hardware bugs.
 ; --
 DEF rWX EQU $FF4B
+DEF WX_OFFSET EQU 7 ; add this to a screen position to get a WX position
```
Comment (ISSOtm) [inline hardware.inc:558]:
```suggestion

DEF WX_OFFSET EQU 7 ; add this to a screen position to get a WX position
```

File: hardware.inc [hunk 4]
```diff
@@ -879,12 +900,15 @@ DEF SCRN_VY_B EQU 32  ; Virtual height of screen in bytes
...
 RSRESET
-DEF OAMA_Y              RB  1   ; y pos
-DEF OAMA_X              RB  1   ; x pos
+DEF OAMA_Y              RB  1   ; y pos plus 16
+DEF OAMA_X              RB  1   ; x pos plus 8
 DEF OAMA_TILEID         RB  1   ; tile id
 DEF OAMA_FLAGS          RB  1   ; flags (see below)
 DEF sizeof_OAM_ATTRS    RB  0
 
+DEF OAMA_Y_OFFSET EQU 16 ; add this to a screen position to get an OAM Y position
```
Comment (ISSOtm) [inline hardware.inc:909]:
```suggestion
DEF OAM_Y_OFS EQU 16 ; add this to a screen-relative Y position to get an OAM Y position
```

File: hardware.inc [hunk 5]
```diff
@@ -879,12 +900,15 @@ DEF SCRN_VY_B EQU 32  ; Virtual height of screen in bytes
...
+DEF OAMA_X_OFFSET EQU 8  ; add this to a screen position to get an OAM X position
```
Comment (ISSOtm) [inline hardware.inc:910]:
```suggestion
DEF OAM_X_OFS EQU 8  ; add this to a screen-relative X position to get an OAM X position
```

**Comment by avivace** [comment]
Why are we suggesting both OFS and OFFSET? @ISSOtm

**Comment by ISSOtm** [comment]
It seemed to me that `WX_OFFSET` was still short enough that fully spelling it out was OK, so I elected not to change it. (I'm not suggesting to correct it, only to add a blank line between it and the register constant, as is done with other "associated constants" throughout the file.)

If you think that's inconsistent, we can change it to `WX_OFS`, I suppose.

**Comment by avivace** [comment]
> It seemed to me that `WX_OFFSET` was still short enough that fully spelling it out was OK, so I elected not to change it. (I'm not suggesting to correct it, only to add a blank line between it and the register constant, as is done with other "associated constants" throughout the file.)
> 
> If you think that's inconsistent, we can change it to `WX_OFS`, I suppose.

You did correct some instances of OFFSET to OFS while you allowed others. I think consistency here should be a priority (then length of the names).

**Comment by ISSOtm** [comment]
Edited accordingly.

**Comment by avivace** [comment]
> Edited accordingly.

Great, thanks!

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**